Core/src/Vertexing/AdaptiveMultiVertexFinder.cpp

```diff
@@ -504,24 +513,36 @@ bool AdaptiveMultiVertexFinder::isMergedVertex(
     const SquareMatrix4& otherCov = otherVtx->fullCovariance();
 
     double significance = 0;
-    if (!m_cfg.do3dSplitting) {
-      const double deltaZPos = otherPos[eZ] - candidatePos[eZ];
-      const double sumVarZ = otherCov(eZ, eZ) + candidateCov(eZ, eZ);
-      if (sumVarZ <= 0) {
-        // TODO FIXME this should never happen
-        continue;
+    if (!m_cfg.doFullSplitting) {
+      if (m_cfg.useTime) {
+        const Vector2 deltaZT = otherPos.tail<2>() - candidatePos.tail<2>();
+        SquareMatrix2 sumCovZT = candidateCov.bottomRightCorner<2, 2>() +
+                                 otherCov.bottomRightCorner<2, 2>();
+        auto sumCovZTInverse = safeInverse(sumCovZT);
+        if (!sumCovZTInverse) {
+          ACTS_ERROR("Vertex z-t covariance matrix is singular.");
```

Comment:
Might be a bit dangerous I suppose, especially in an athena context.

Reply:
Two thoughts on this from my side
1. we try it as this should in principle never happen. but this is potentially risky as it might pop up in months
2. we just lower the log level and propagate the `Result` error

Reply:
Nightly from last night is broken, so cannot run regular validation job right now. We'd have to wait until tomorrow for the automatic way. Otherwise we can run a manual job.

Reply:
let's do it tomorrow